lib/auth/iam.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export default class IAMAuth {
+
+  /**
+   * IAM credentials.
+   *
+   * @param {string=} selector the iam authority selector
```

Comment:
I want to come back around and make sure I have all of the types correct after this CL, and then we can start removing some of these.  Either way - pretty sure tsdoc actually uses them.